src/main/kotlin/com/chattriggers/ctjs/minecraft/libs/ChatLib.kt

```diff
@@ -420,6 +421,41 @@ object ChatLib {
         return message.toString().replace("(?<!\\\\)&(?![^0-9a-fk-or]|$)".toRegex(), "\u00a7")
     }
 
+    /**
+     * Gets the position of the caret in the chat's input box.
+     * @return the zero-index caret position, -1 if not in chat
+     */
+    @JvmStatic
+    fun getCaretPosition(): Int {
+        if (Client.getMinecraft().currentScreen is GuiChat) {
+            return (Client.getMinecraft().currentScreen as GuiChat).inputField.cursorPosition
+        } else {
+            return -1
+        }
```

Comment:
```suggestion
        return (Client.getMinecraft().currentScreen as? GuiChat)?.inputField?.cursorPosition ?: -1
```

Similar suggestions for the other methods as well (though I can't remember if this syntax works for assignment)